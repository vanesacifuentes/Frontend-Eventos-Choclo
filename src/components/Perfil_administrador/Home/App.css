.content{
    display: flex;
    font-family: 'Verdana', sans-serif;
    text-decoration-color: black;
}

