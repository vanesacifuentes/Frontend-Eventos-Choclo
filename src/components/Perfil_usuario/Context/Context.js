import { createContext, useState } from "react";









const Context = createContext();

const UserProvider = ({ children }) => {


  let stringUsuario = localStorage.getItem('usuario')

  const [usuario, setUser] = useState(stringUsuario && JSON.parse(stringUsuario))

  const [evento, setEvento] = useState({})

  function isObjetoVacio(obj) {//valida si un objeto esta vacio :: retorna true o false
    return Object.getOwnPropertyNames(obj).length === 0;
  }

  function setearEvento(evento) {
    console.log('setEvento')
    setEvento(evento)
  }
  async function userF(u) {

    let objUsuario = JSON.parse(u)


    if (isObjetoVacio(objUsuario)) {
      return false

    } else {
      localStorage.setItem('usuario', JSON.stringify(objUsuario))
      return true
    }
  }



  return (
    <Context.Provider value={{ usuario, userF, evento, setearEvento }}>

      {children}

    </Context.Provider>
  )
}



export { UserProvider }
export default Context