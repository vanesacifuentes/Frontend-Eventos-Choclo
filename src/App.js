import Home from './components//Perfil_usuario/Home/Home';
import Eventos from './components//Perfil_usuario/Eventos/Eventos'
import Contacto from './components//Perfil_usuario/Contacto/Contacto'
import Register_User from './components//Perfil_usuario/Registro/register_user';
import LoginUser from './components//Perfil_usuario/Login/login-user';
import Boleteria from './components/Perfil_usuario/Boleteria/Boleteria';
import Compra_boleta from './components/Perfil_usuario/Boleteria/Compra_boleta';
import Context from './components/Perfil_usuario/Context/Context';

import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import { useContext } from 'react';











function App({inicio = <Home />}) {

  const {usuario} = useContext(Context)

  

  return (
    <>

            
            <Router>
              <div>
                <Switch>

                  <Route exact path='/' > {inicio} </Route>

                  <Route exact path='/eventos' > <Eventos /> </Route>
                  <Route exact path='/contacto' > <Contacto /> </Route>
                  <Route exact path='/loginuser'>  <LoginUser  /></Route>
                  <Route exact path='/registeruser'> <Register_User /> </Route>
                  <Route exact path='/boleteria'> <Boleteria /> </Route>
                  <Route exact path='/compra_boleta'> <Compra_boleta /> </Route>              


                </Switch>
              </div>

            </Router>
          
      </>
  );
}

export default App;
