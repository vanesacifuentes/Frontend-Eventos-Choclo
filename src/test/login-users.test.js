import Enzyme, { ShallowWrapper,mount } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { createSerializer } from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))

import LoginUser from '../components/perfil_usuario/Login/login-user'
//import Context from '../components/perfil_usuario/Context/Context'
//import App from '../index'
import React, { useContext } from 'react';
import { shallow } from 'enzyme'
import { Router, Switch } from 'react-router'
import Register_User from '../components/Perfil_usuario/Registro/register_user'
import Context from '../components/Perfil_usuario/Context/Context'
import App from '../App'
import { UserProvider } from '../components/Perfil_usuario/Context/Context'

describe('Pruebas sobre el componente Login usuario', () => {

    //Snapshot
    /*test('Probando con enzyme', ()=>{
        const wrapper = shallow(<LoginUser/>)
        expect(wrapper).toMatchSnapshot()
    })*/

    const wrapper = mount(
    
        <UserProvider>

            <App inicio={<LoginUser/>}/>
        </UserProvider>
       
 
        
    )

    test('Verificar mensajes de error en input usuario', () => {
        
        expect(wrapper).toMatchSnapshot()
        const input = wrapper.find('#usuario').simulate('change', { target: { name: 'usuario' , value: 'Hello' } })
        
        console.log(wrapper.debug())
        //expect(input.debug()).toEqual('Hello')

        wrapper.find('#boton1').simulate('click')
        
            expect(wrapper.find('.invalid-feedback').at(0).text()).toEqual('Ingrese el usuario')
            expect(wrapper.find('.invalid-feedback').at(1).text()).toEqual('Ingrese la contraseÃ±a')

       // }
        
    })

   // test('Verificar mensajes de error en input contrasena', ()=>{
        //expect(wrapper).toMatchSnapshot()
        //const botonIngresar = wrapper.find('#boton1').simulate('click')
        

    //})


})
