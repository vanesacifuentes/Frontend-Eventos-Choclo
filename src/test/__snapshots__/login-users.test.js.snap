// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente Login usuario Verificar mensajes de error en input usuario 1`] = `
<div>
  Array [
    " ",
    <div
      className="container-fluid "
    >
      <div
        className=" d-flex justify-content-center align-items-center "
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h1>
            Inicio
          </h1>
        </a>
      </div>
      <div
        className="modal-dialog text-center mt-1"
      >
        <div
          className="col-sm-9 main-section"
        >
          <div
            className="modal-content"
          >
            <div
              className="col-12 user-img"
            >
              <i
                className="mt-2 fas fa-users fa-6x"
                style={
                  Object {
                    "color": "#ffff",
                  }
                }
              />
              <h3
                className="h3 mb-2"
              >
                ¡Bienvenido!
              </h3>
              <h6
                className="h6 mb-3 align-items-start"
              >
                Por favor autentíquese para ingresar
              </h6>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group "
                  id="#user-group"
                >
                  <label>
                    Usuario
                  </label>
                  <input
                    className=""
                    id="usuario"
                    name="usuario"
                    onChange={[Function]}
                    placeholder="Usuario"
                    type="text"
                  />
                  <div
                    className="h5 mb-2 invalid-feedback"
                  >
                    Ingrese el usuario
                  </div>
                </div>
                <div
                  className="form-group"
                  id="contrasena-group"
                >
                  <label>
                    Contraseña
                  </label>
                  <input
                    className=""
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Contraseña"
                    type="password"
                  />
                  <div
                    className="h5 mb-2 invalid-feedback"
                  >
                    Ingrese la contraseña
                  </div>
                </div>
                <div
                  className="row justify-content-center align-items-center mt-3 mb-3"
                >
                  <button
                    className="btn col-3 mx-3"
                    id="boton1"
                    type="submit"
                  >
                    Ingresar
                  </button>
                  <button
                    className="btn col-4"
                    id="boton2"
                    onClick={[Function]}
                    type="button"
                  >
                    Registrarse
                  </button>
                </div>
                <h6
                  className=" h6 "
                >
                  ¿Olvidó su contraseña?
                </h6>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>,
    " ",
  ]
</div>
`;
